src/reducers/index.js

```diff
@@ -10,7 +13,9 @@ import userReducer from './user.reducer';
 const rootReducer = combineReducers({
   user: userReducer,
   annotations: trailAnnotationsReducer,
-  trails: trailsReducer
+  trails: trailsReducer,
+  map:mapReducer,
+  tabs:tabsReducer
```

Comment:
The request to switch the tab is sent from the reportProblemContainer to the trailMenuContainer.  When you capture the photo in the reportProblemContainer, it sends it to the server, then asks the map container to pan the view to where the photo was taken, and also asks the menu container to switch tabs.

The happens in reportProblem component:
         obj.props.saveAnnotation({'lat':obj.state.lat, 'lng': obj.state.lng,'photo':fileData});
         obj.props.fetchAnnotations();
         obj.props.panMapActionCreator(obj.state.lat, obj.state.lng);
         obj.props.switchTabActionCreator('location');

Using the reducers does feel cleaner to me than passing  a reference to map and tab menu via props to the report problem container.  It's nice to be able to, say, pan the map and switch a tab and know all actions are handled via the reducer mechanism.  Also, I'd consider the current coordinates of the map (it's panned position) and currently selected menu tab to be application state.

But that's just based on my very limited knowledge of the stuff.  So perhaps when we do the demo you can give me a better idea of how to distinguish what state should be changed via reducers and what shouldn't.